pennylane/devices/preprocess.py

```diff
@@ -205,6 +211,20 @@ def mid_circuit_measurements(
     )
 
 
+def _is_conditional_mcm(operation: qml.operation.Operator) -> bool:
+    """Returns True if the operation is a mid-circuit measurement nested inside a Conditional,
+    and False otherwise."""
+    if isinstance(operation, qml.ops.Conditional):
+        if isinstance(operation.base, qml.measurements.MidMeasureMP):
+            return True
+    return False
+
+
+@lru_cache()
```

Comment:
```suggestion
@lru_cache(maxsize=2)
```
Always good to set a limit to these incase they infinitely capture and store. Assuming 2 different inputs is probably fine (I'd expect the same input most times), but happy to go with a different parameter here if it seems sensible.